File: 3838.weighted-word-mapping/solution.kt
class Solution {
    fun mapWordWeights(words: Array<String>, weights: IntArray): String {
        return words.joinToString("") { word ->
            val weight = word.sumOf { weights[it - 'a'] }
            val mappedChar = 'z' - weight % 26
                
           "$mappedChar"
        }
    }
}

File: 3838.weighted-word-mapping/solution.go
func mapWordWeights(words []string, weights []int) string {
    res := make([]byte, 0, len(words))

    for _, word := range words {
        weight := 0
        for i := range word {
            weight += weights[word[i] - 'a']
        }
        ch := 'z' - byte(weight % 26)
        res = append(res, ch)
    }

    return string(res)
}